node-gyp fails on node 12.5.0; node-gyp 5.0.3

Hii, I wanna try your library, it seems interesting. However I got this error when building.

../src/main.cpp:308:61: error: too few arguments to function call, single argument 'context' was not specified
      Nan::New<v8::FunctionTemplate>(OpenFile)->GetFunction());
      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
/Users/rikvermeer/Library/Caches/node-gyp/12.5.0/include/node/v8.h:5948:3: note: 'GetFunction' declared here
  V8_WARN_UNUSED_RESULT MaybeLocal<Function> GetFunction(

--- a/.travis.yml
+++ b/.travis.yml
@@ -1,7 +1,8 @@
 dist: trusty
 language: node_js
 node_js:
-  - "10.16.1"
+  - '10'
+  - '12'
 before_install:
   - sudo rm -f /usr/local/bin/python-config
   - sudo ln -s /opt/python/3.6.3/bin/python3-config /usr/local/bin/python-config

--- a/src/helpers.cpp
+++ b/src/helpers.cpp
@@ -3,7 +3,7 @@
 PyObject *BuildPyDict(v8::Local<v8::Value> arg)
 {
   auto obj = arg.As<v8::Object>();
-  auto keys = obj->GetOwnPropertyNames();
+  auto keys = obj->GetOwnPropertyNames(Nan::GetCurrentContext()).ToLocalChecked();
   PyObject *dict = PyDict_New();
   for (unsigned int i = 0; i < keys->Length(); i++)
   {
@@ -30,7 +30,7 @@ PyObject *BuildPyDict(v8::Local<v8::Value> arg)
     }
     else if (val->IsBoolean())
     {
-      long b = val->BooleanValue();
+      long b = val->BooleanValue(Nan::GetCurrentContext()).FromJust();
       PyDict_SetItem(dict, pyKey, PyBool_FromLong(b));
     }
     else if (val->IsDate())
@@ -95,7 +95,7 @@ PyObject *BuildPyArray(v8::Local<v8::Value> arg)
     }
     else if (element->IsBoolean())
     {
-      long b = element->BooleanValue();
+      bool b = element->BooleanValue(Nan::GetCurrentContext()).FromJust();
       PyList_SetItem(list, i, PyBool_FromLong(b));
     }
     else if (element->IsDate())
@@ -160,7 +160,7 @@ PyObject *BuildPyArgs(const Nan::FunctionCallbackInfo<v8::Value> &args)
     }
     else if (arg->IsBoolean())
     {
-      long b = arg->BooleanValue();
+      long b = arg->BooleanValue(Nan::GetCurrentContext()).FromJust();
       PyTuple_SetItem(pArgs, i - 1, PyBool_FromLong(b));
     }
     else if (arg->IsDate())

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -243,7 +243,7 @@ class CallWorker : public Nan::AsyncWorker {
     PyObject *pFunc;
 };
 
-NAN_METHOD(callAsync) {
+NAN_METHOD(call) {
   if (info.Length() == 0 || !info[0]->IsString()) {
     Nan::ThrowError("First argument to 'call' must be a string");
     return;
@@ -283,40 +283,12 @@ NAN_METHOD(callAsync) {
 
 void Initialize(v8::Local<v8::Object> exports)
 {
-  // exports->Set(
-  //     Nan::New("call").ToLocalChecked(),
-  //     Nan::New<v8::FunctionTemplate>(CallAsync)->GetFunction());
-
-  NAN_EXPORT(exports, callAsync);
-
-  // exports->Set(
-  //     Nan::New("dlOpen").ToLocalChecked(),
-  //     Nan::New<v8::FunctionTemplate>(DLOpen)->GetFunction());
+  NAN_EXPORT(exports, call);
   NAN_EXPORT(exports, dlOpen);
-
-  // exports->Set(
-  //     Nan::New("startInterpreter").ToLocalChecked(),
-  //     Nan::New<v8::FunctionTemplate>(StartInterpreter)->GetFunction());
   NAN_EXPORT(exports, startInterpreter);
-
-  // exports->Set(
-  //     Nan::New("stopInterpreter").ToLocalChecked(),
-  //     Nan::New<v8::FunctionTemplate>(StopInterpreter)->GetFunction());
   NAN_EXPORT(exports, stopInterpreter);
-
-  // exports->Set(
-  //     Nan::New("appendSysPath").ToLocalChecked(),
-  //     Nan::New<v8::FunctionTemplate>(appendSysPath)->GetFunction());
   NAN_EXPORT(exports, appendSysPath);
-
-  // exports->Set(
-  //     Nan::New("openFile").ToLocalChecked(),
-  //     Nan::New<v8::FunctionTemplate>(OpenFile)->GetFunction());
   NAN_EXPORT(exports, openFile);
-
-  // exports->Set(
-  //     Nan::New("eval").ToLocalChecked(),
-  //     Nan::New<v8::FunctionTemplate>(Eval)->GetFunction());
   NAN_EXPORT(exports, eval);
 }
 

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -16,7 +16,7 @@
 
 PyObject *pModule;
 
-void DLOpen(const Nan::FunctionCallbackInfo<v8::Value> &args)
+void dlOpen(const Nan::FunctionCallbackInfo<v8::Value> &args)
 {
   #ifdef _WIN32
     Nan::ThrowError("dlOpen does not work in windows");
@@ -34,7 +34,7 @@ void DLOpen(const Nan::FunctionCallbackInfo<v8::Value> &args)
   #endif
 }
 
-void StartInterpreter(const Nan::FunctionCallbackInfo<v8::Value> &args)
+void startInterpreter(const Nan::FunctionCallbackInfo<v8::Value> &args)
 {
   if (args.Length() == 1 && args[0]->IsString()) {
     Nan::Utf8String pathString(args[0]);
@@ -46,7 +46,7 @@ void StartInterpreter(const Nan::FunctionCallbackInfo<v8::Value> &args)
   Py_Initialize();
 }
 
-void StopInterpreter(const Nan::FunctionCallbackInfo<v8::Value> &args)
+void stopInterpreter(const Nan::FunctionCallbackInfo<v8::Value> &args)
 {
   auto isInitialized = Py_IsInitialized();
   if (isInitialized == 0) return;
@@ -55,7 +55,7 @@ void StopInterpreter(const Nan::FunctionCallbackInfo<v8::Value> &args)
   pModule = NULL;
 }
 
-void AppendSysPath(const Nan::FunctionCallbackInfo<v8::Value> &args)
+void appendSysPath(const Nan::FunctionCallbackInfo<v8::Value> &args)
 {
   if (args.Length() == 0 || !args[0]->IsString()) {
     Nan::ThrowError("Must pass a string to 'appendSysPath'");
@@ -73,7 +73,7 @@ void AppendSysPath(const Nan::FunctionCallbackInfo<v8::Value> &args)
   free(appendPathStr);
 }
 
-void OpenFile(const Nan::FunctionCallbackInfo<v8::Value> &args)
+void openFile(const Nan::FunctionCallbackInfo<v8::Value> &args)
 {
   if (args.Length() == 0 || !args[0]->IsString()) {
     Nan::ThrowError("Must pass a string to 'openFile'");
@@ -97,7 +97,7 @@ void OpenFile(const Nan::FunctionCallbackInfo<v8::Value> &args)
   }
 }
 
-void Eval(const Nan::FunctionCallbackInfo<v8::Value> &args)
+void eval(const Nan::FunctionCallbackInfo<v8::Value> &args)
 {
   if (args.Length() == 0 || !args[0]->IsString()) {
     Nan::ThrowError("Must pass a string to 'eval'");
@@ -243,7 +243,7 @@ class CallWorker : public Nan::AsyncWorker {
     PyObject *pFunc;
 };
 
-NAN_METHOD(CallAsync) {
+NAN_METHOD(callAsync) {
   if (info.Length() == 0 || !info[0]->IsString()) {
     Nan::ThrowError("First argument to 'call' must be a string");
     return;
@@ -283,33 +283,41 @@ NAN_METHOD(CallAsync) {
 
 void Initialize(v8::Local<v8::Object> exports)
 {
-  exports->Set(
-      Nan::New("call").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(CallAsync)->GetFunction());
+  // exports->Set(
+  //     Nan::New("call").ToLocalChecked(),
+  //     Nan::New<v8::FunctionTemplate>(CallAsync)->GetFunction());
 
-  exports->Set(
-      Nan::New("dlOpen").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(DLOpen)->GetFunction());
+  NAN_EXPORT(exports, callAsync);
 
-  exports->Set(
-      Nan::New("startInterpreter").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(StartInterpreter)->GetFunction());
+  // exports->Set(
+  //     Nan::New("dlOpen").ToLocalChecked(),
+  //     Nan::New<v8::FunctionTemplate>(DLOpen)->GetFunction());
+  NAN_EXPORT(exports, dlOpen);
 
-  exports->Set(
-      Nan::New("stopInterpreter").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(StopInterpreter)->GetFunction());
+  // exports->Set(
+  //     Nan::New("startInterpreter").ToLocalChecked(),
+  //     Nan::New<v8::FunctionTemplate>(StartInterpreter)->GetFunction());
+  NAN_EXPORT(exports, startInterpreter);
 
-  exports->Set(
-      Nan::New("appendSysPath").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(AppendSysPath)->GetFunction());
+  // exports->Set(
+  //     Nan::New("stopInterpreter").ToLocalChecked(),
+  //     Nan::New<v8::FunctionTemplate>(StopInterpreter)->GetFunction());
+  NAN_EXPORT(exports, stopInterpreter);
 
-  exports->Set(
-      Nan::New("openFile").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(OpenFile)->GetFunction());
+  // exports->Set(
+  //     Nan::New("appendSysPath").ToLocalChecked(),
+  //     Nan::New<v8::FunctionTemplate>(appendSysPath)->GetFunction());
+  NAN_EXPORT(exports, appendSysPath);
 
-  exports->Set(
-      Nan::New("eval").ToLocalChecked(),
-      Nan::New<v8::FunctionTemplate>(Eval)->GetFunction());
+  // exports->Set(
+  //     Nan::New("openFile").ToLocalChecked(),
+  //     Nan::New<v8::FunctionTemplate>(OpenFile)->GetFunction());
+  NAN_EXPORT(exports, openFile);
+
+  // exports->Set(
+  //     Nan::New("eval").ToLocalChecked(),
+  //     Nan::New<v8::FunctionTemplate>(Eval)->GetFunction());
+  NAN_EXPORT(exports, eval);
 }
 
 extern "C" NODE_MODULE_EXPORT void

